libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -330,88 +190,175 @@ void ies_enkf_updateA(
-
-    const matrix_type *dataE = ies_enkf_data_getE(data);
-
-    int m = -1; // counter for currently active measurements
-    for (int iobs = 0; iobs < obs_size_msk; iobs++) {
-        if (bool_vector_iget(obs_mask, iobs)) {
-            m++;
-            fprintf(log_fp, "ies_enkf_linalg_extract_active iobs=%d m=%d)\n",
-                    iobs, m);
-            int i = -1;
-            for (int iens = 0; iens < ens_size_msk; iens++) {
-                if (bool_vector_iget(ens_mask, iens)) {
-                    i++;
-                    matrix_iset_safe(E, m, i, matrix_iget(dataE, iobs, iens));
+    int ens_size = matrix_get_columns(
+        Yin); // Number of active realizations in current iteration
+    int state_size = matrix_get_rows(A);
+
+    int iteration_nr = ies_enkf_data_inc_iteration_nr(data);
+    FILE *log_fp = ies_enkf_data_open_log(data);
+
+    const double ies_steplength =
+        ies_enkf_config_calculate_steplength(ies_config, iteration_nr);
+
+    ies_enkf_data_update_state_size(data, state_size);
+
+    /*
+      Counting number of active observations for current iteration. If the
+      observations have been used in previous iterations they are contained in
+      data->E0. If they are introduced in the current iteration they will be
+      augmented to data->E.
+    */
+    ies_enkf_data_store_initialE(data, Ein);
+    ies_enkf_data_augment_initialE(data, Ein);
+    ies_enkf_data_store_initialA(data, A);
+
+    /*
+     * Re-structure input matrices according to new active obs_mask and ens_size.
+     * Allocates the local matrices to be used.
+     * Copies the initial measurement perturbations for the active observations into the current E matrix.
+     * Copies the inputs in D, Y and R into their local representations
+     */
+    matrix_type *Y = matrix_alloc_copy(Yin);
+    matrix_type *R = matrix_alloc_copy(Rin);
+    matrix_type *E = ies_enkf_alloc_activeE(data);
+    matrix_type *D = matrix_alloc_copy(Din);
+    matrix_type *X = matrix_alloc(ens_size, ens_size);
+
+    /* Subtract new measurement perturbations              D=D-E    */
+    matrix_inplace_sub(D, Ein);
+    /* Add old measurement perturbations */
+    matrix_inplace_add(D, E);
+
+    double costf;
+    ies_enkf_initX__(
+        ies_enkf_config_get_ies_aaprojection(ies_config) ? A : nullptr, Y, R, E,
+        D, X, ies_enkf_config_get_ies_inversion(ies_config),
+        ies_enkf_config_get_truncation(ies_config),
+        ies_enkf_config_get_enkf_subspace_dimension(ies_config),
+        ies_enkf_config_get_ies_aaprojection(ies_config), data, ies_steplength,
+        iteration_nr, &costf);
+    logger->info("IES  iter:{} cost function: {}", iteration_nr, costf);
+
+    /* COMPUTE NEW ENSEMBLE SOLUTION FOR CURRENT ITERATION  Ei=A0*X (Line 11)*/
+    int m_ens_size = std::min(ens_size - 1, 16);
+    int m_state_size = std::min(state_size - 1, 3);
+    matrix_pretty_fprint_submat(A, "A^f", "%11.5f", log_fp, 0, m_state_size, 0,
```

Comment:
This used to be behind a `dbg` flag, is it a leftover from debugging? Same a few lines below.

Reply:
OK